platform/linux-generic/odp_packet_io.c

```diff
@@ -1288,12 +1288,44 @@ odp_time_t odp_pktin_ts_from_ns(odp_pktio_t hdl, uint64_t ns)
 		return ODP_TIME_NULL;
 	}
 
-	if (entry->s.ops->pktin_ts_from_ns)
-		return entry->s.ops->pktin_ts_from_ns(entry, ns);
+	if (entry->s.ops->pktio_ts_from_ns)
+		return entry->s.ops->pktio_ts_from_ns(entry, ns);
 
 	return odp_time_global_from_ns(ns);
 }
 
+odp_time_t odp_pktio_time(odp_pktio_t hdl, odp_time_t *global_ts)
+{
+	pktio_entry_t *entry;
+
+	entry = get_pktio_entry(hdl);
+	if (entry == NULL) {
+		ODP_DBG("pktio entry %d does not exist\n", hdl);
+		return ODP_TIME_NULL;
+	}
+
+	/* Callback if present */
+	if (entry->s.ops->pktio_time)
+		return entry->s.ops->pktio_time(entry, global_ts);
+
+	/* By default both Packet IO time source and
+	 * global time source are same.
+	 */
+	if (global_ts)
+		*global_ts = odp_time_global();
+	return *global_ts;
```

Comment:
This is a null pointer dereference if the caller is not interested in global time.